api/clients/codecs/blob_codec.go

```diff
@@ -1,48 +1,73 @@
+// distribute blobs in Eval form, then the data returned from this function is the final blob representation.
+func EncodePayload(payload []byte) []byte {
+	payloadHeader := make([]byte, 32)
+	// first byte is always 0 to ensure the payloadHeader is a valid bn254 element
+	payloadHeader[1] = byte(DefaultBlobEncoding) // encode version byte
 
-func BlobEncodingVersionToCodec(version BlobEncodingVersion) (BlobCodec, error) {
-	switch version {
-	case DefaultBlobEncoding:
-		return DefaultBlobCodec{}, nil
-	default:
-		return nil, fmt.Errorf("unsupported blob encoding version: %x", version)
-	}
+	// encode payload length as uint32
+	binary.BigEndian.PutUint32(
+		payloadHeader[2:6],
+		uint32(len(payload))) // uint32 should be more than enough to store the length (approx 4gb)
+
+	// encode payload modulo bn254
+	// the resulting bytes subsequently may be treated as the evaluation of a polynomial
+	polynomialEval := codec.ConvertByPaddingEmptyByte(payload)
+
+	encodedPayload := append(payloadHeader, polynomialEval...)
+
+	return encodedPayload
 }
 
-func GenericDecodeBlob(data []byte) ([]byte, error) {
-	if len(data) <= 32 {
-		return nil, fmt.Errorf("data is not of length greater than 32 bytes: %d", len(data))
-	}
-	// version byte is stored in [1], because [0] is always 0 to ensure the codecBlobHeader is a valid bn254 element
-	// see https://github.com/Layr-Labs/eigenda/blob/master/api/clients/codecs/default_blob_codec.go#L21
-	// TODO: we should prob be working over a struct with methods such as GetBlobEncodingVersion() to prevent index errors
-	version := BlobEncodingVersion(data[1])
-	codec, err := BlobEncodingVersionToCodec(version)
-	if err != nil {
-		return nil, err
+// DecodePayload accepts bytes representing an encoded payload, and returns the decoded payload
+//
+// This function expects the parameter bytes to be a polynomial in Eval form. In other words, if blobs in the system
+// are being distributed in Coeff form, a blob must be FFTed prior to being passed into the function.
+func DecodePayload(encodedPayload []byte) ([]byte, error) {
+	if len(encodedPayload) < 32 {
+		return nil, fmt.Errorf("encoded payload does not contain 32 header bytes, meaning it is malformed")
 	}
 
-	data, err = codec.DecodeBlob(data)
+	payloadLength := binary.BigEndian.Uint32(encodedPayload[2:6])
+
+	// decode raw data modulo bn254
+	nonPaddedData := codec.RemoveEmptyByteFromPaddedBytes(encodedPayload[32:])
+
+	reader := bytes.NewReader(nonPaddedData)
+	payload := make([]byte, payloadLength)
+	readLength, err := reader.Read(payload)
 	if err != nil {
-		return nil, fmt.Errorf("unable to decode blob: %w", err)
+		return nil, fmt.Errorf(
+			"failed to copy unpadded data into final buffer, length: %d, bytes read: %d",
+			payloadLength, readLength)
+	}
+	if uint32(readLength) != payloadLength {
+		return nil, fmt.Errorf("data length does not match length prefix")
```

Comment:
I'd prefer if this was in a self-contained function. Maybe we can change RemoveEmptyByteFromPaddedBytes to take a second `len` argument?